README.md

```diff
@@ -239,6 +240,92 @@ This already goes with how CSP is currently enforcing its rules canonically in t
 
 `X-frames` and `frame-src` allow controlling what is allowed to be loaded into an `iframe`. Regardless of their use to deny loading content in iframes, ways to crete new same origin realms remain available, such as creating an `about:blank` iframe, opening a new tab using `open` API and more.
 
+## Consideration
+
+Some important points to consider
+
+### Privacy
+
+There isn't much to take into consideration in regards with privacy.
+
+### Security
+
+Here are some security risks to take into consideration when following this spec for implementation purposes:
+
+#### Universal XSS
+
+Since this proposal focuses on the thin line between same and cross origins, it may put SOP in danger if not implemented correctly.
+
+Since the proposed CSP directive introduces a remote resource to be treated as JavaScript code to execute in all realms that their origin corresponds to the origin of the top most document, if telling same from cross origin goes wrong, the browser may execute the JavaScript code in a cross realm rather than an origin realm, resulting in a Universal XSS effectively.
+
+So if website `attacker.com` finds such a flaw in the implementation, they can easily initiate a Universal XSS against `facebook.com` by taking the following steps:
+
+```html
+<script src="/x.js">
+   // browser got tricked into running RIC script under a cross origin:
+   alert(document.cookie); // will execute under facebook.com
+</script>
+
+<!-- Content-Security-Policy: init-realm: /x.js -->
+
+<html location="//attacker.com">
+<script>
+   createAnIframeThatBypassesRIC({src: '//facebook.com'});
+</script>
+</html>
+```
+
+Therefore, it is crucial to make sure that the check in which it is determined whether the origin is the same as the top most document or not is firm and trustworthy and takes all possible edge cases into account.
+
+While this sounds alarming, since this proposal advises to tap into existing places in which such information is safely determined, it is fair to believe this can be implemented safely just as well.
+
+#### Escalation to Code Execution
```

Comment:
That's an argument against using a header here.. Developer already (should) defend against script execution in their documents, but not necessarily in their headers